tests/test_publication_info.py

```diff
@@ -0,0 +1,102 @@
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+from invenio.testsuite import InvenioTestCase, make_test_suite, run_test_suite
+from flask import current_app
+
+
+class PubInfoTests(InvenioTestCase):
+
+    def setUp(self):
+        self.pub_info = current_app.jinja_env.filters['publication_info']
+        self.pub_info_unique_field = {"publication_info": [{u'journal_volume': u'1503', u'journal_title': u'JCAP', u'page_artid': u'044', u'year': 2015}]
+                                      }
+        self.pub_info_multiple_fields = {"publication_info": [{u'journal_volume': u'1401', u'journal_title': u'JHEP', u'journal_recid': 1213103,
+                                                               u'page_artid': u'163', u'year': 2014}, {u'note': u'Erratum', u'journal_recid': 1213103,
+                                                                                                       u'page_artid': u'014', u'journal_title': u'JHEP', u'journal_volume': u'1501', u'year': 2015}]}
+        self.pub_info_freetext_field = {u'publication_info': [
+            {u'pubinfo_freetext': u'Current Science Vol. 109 (N0. 12, 25 December 2015), 2220-2229\n (2015)'}]}
+        self.conf_info_first_case = {"publication_info": [{u'journal_volume': u'C050318', u'journal_title': u'eConf', 
+                                                           u'page_artid': u'0106', u'year': 2005},
+                                                          {u'parent_recid': 706120,
+                                                           u'cnum': u'C05-03-18', u'conference_recid': 976391,
+                                                           u'confpaper_info': u'To appear in the proceedings of'}]}
+        self.conf_info_second_case = {"publication_info": [
+            {u'parent_recid': 720114, u'cnum': u'C05-08-14.1', u'confpaper_info': u'Invited talk at'}]}
+        self.conf_info_second_case_with_page = {"publication_info": [
+            {u'parent_recid': 1402672, u'cnum': u'C15-03-14', u'conference_recid': 1331207, u'page_artid': u'515-518'}]}
+        self.conf_info_third_case = {
+            "publication_info": [{u'cnum': u'C15-06-29.3', u'conference_recid': 1346335}]}
+
+        self.pub_info_unique = {"pub_info": [u'<i>JCAP</i> 1503 (2015) 044']}
+        self.pub_info_multiple = {
+            'pub_info': [u'<i>JHEP</i> 1401 (2014) 163', u'<i>JHEP</i> 1501 (2015) 014']}
+        self.pub_info_freetext = {'pub_info': [
+            u'Current Science Vol. 109 (N0. 12, 25 December 2015), 2220-2229\n (2015)']}
+        self.conf_info_conf_and_par = {
+            'conf_info': u'(<a href="/record/706120">Proceedings</a> of <a href="/record/976391">  2005 International Linear Collider Workshop (LCWS 2005)</a>)'}
+        self.conf_info_only_par = {
+            'conf_info': u'\n \n <a href="/record/720114">Proceedings</a> of 2005 International Linear Collider Physics and Detector Workshop and 2nd ILC Accelerator Workshop (Snowmass 2005)\n \n'}
```

Comment:
It is added because of the macros file that we are using. This happens because I use {% macro conf_with_pub_info(parent_recid, conference_recid, conference_title) %} and not sth like 
{%- macro conf_with_pub_info(parent_recid, conference_recid, conference_title) -%}. I can remove it actually in most of them except the case that I am using this 
```
{% macro conference_only(conference_recid, conference_title, pub_info = false) %}
  {% if pub_info %}
    <br/>
  {% endif %}
  Contribution to <a href="/record/{{ conference_recid }}">{{ conference_title }}</a>
{% endmacro %}
```
because I need to break the line in this one.